edgedb-protocol/src/value.rs

```diff
@@ -626,6 +652,23 @@ mod test {
         );
     }
 
+    #[test]
+    fn display() {
+        let cases = [
+            0,
+            1,
+            -1,
+            1_0000,
+            -1_0000,
+            1_2345_6789,
+            i64::MAX,
+            i64::MIN,
+        ];
```

Comment:
Can you also add a simple fuzz-test: generate like 100 random ints in the i64::MIN..i64::MAX range in addition to the hardcoded cases. Over the years of dealing with rendering code like this such tests helped me a number of times.

Reply:
Added some randomized tests.

Btw you might want to tweak `gen_i64` so it doesn't only favor short numbers, but also numbers with trailing zeros, since your `BigInt` representation treats those as special.

Reply:
Yeah, that's what I did in https://github.com/edgedb/edgedb-python/blob/ddc885d043c28bd30fd0ec7c5742e7600010b1e6/tests/test_async_fetch.py#L565-L638